### DISCUSSION TITLE
Is compiling Fable to Rust still actively being developed?

### DISCUSSION THREAD

**Discussion by general-rishkin** [fable-compiler/Fable#3938] [General]
I only ask because it seems there has not been any real progress since perhaps 2022/early 2023 and because I would like to use F# compiled to Rust for high performance computing. 
If this is going to be a thing then I can forge ahead confidently.

**Comment by ncave** [comment]
@general-rishkin

> it seems there has not been any real progress since perhaps 2022/early 2023

The short answer is yes, there have been multiple improvements since then, but just to understand better, do you mind clarifying how you reached that assessment, and what are you looking for to be considered as real progress?
Is it a particular feature you're looking for, or more documentation?

**Reply by general-rishkin** [reply]
Hi ncave

I guess I am just looking for reassurance that I can write my code in F# and compile it to rust with confidence and make this a consistent workflow. My work involves a lot of hpc simulations, which F# is not really the 'best' tool for. 'But', I really do not want to give up F# as it has the best ergonomics of all languages out there.

**Comment by ncave** [comment]
@general-rishkin

Hey, thanks for your interest in the F# to Rust development.
We all share the love for the beautiful F# language, that's why we're here.

First, apologies for giving unsolicited advice.

I won't pretend to know what your circumstances look like, but if you want confidence and predictability from your dev tooling (and who doesn't), I would recommend perhaps looking at using the official .NET F# compiler directly without transpiling to another language. Nowadays you can compile F# to native code, and even managed F# code performance has improved a lot in F# 8.0. There are some great [videos](https://www.youtube.com/watch?v=NZ5Lwzrdoe8) on YouTube how to write 100x faster F# code for simulations, perhaps you've seen it already.

That said, as one of the contributors of the Fable F# to Rust target, I can say we're making steady progress. I can't go as far as to say it's stable yet, as it's still in active development and we could always use more eyes on it, or people to actually try it, or more people contributing. As an unsupported purely open-source project done on our spare time, we obviously don't have the resources that the official F# compiler has to throw at it.

So if using the Fable F# to Rust transpiler as part of your flow makes sense and you're ok with using pre-release bits, by all means try the Fable to Rust code generation, just keep in mind it's still being developed, as you can see from the commits.

As always, any contributions, PRs, issues or ideas (or just feedback) are always welcome, and no contribution is too small.

Have an awesome day!

**Reply by general-rishkin** [reply]
Thanks @ncave. I will carry out experiments with the F# to Rust transpiler.

**Reply by thautwarm** [reply]
I'd like to work on this (Fable Rust target). Used to use Fable a lot and developed much with Rust in recent years.

**Reply by andrewcurtis** [reply]
I'd also like to contribute , but noticed there are no issues related to the rust target (nor even labels in the issue tracker). Can we request some info? Apologies if its there and I missed it...

**Reply by ncave** [reply]
@thautwarm @andrewcurtis

Thank you for your interest in the F# to Rust development.

As stated above, any contributions of PRs, issues or ideas (or just feedback) are always welcome, and no contribution is too small. I'm sure once you start using the F# to Rust transpiler, there will be many questions or improvements you can suggest.

Here is the Rust feature tracking issue, lots of options for contributing, big and small:
https://github.com/fable-compiler/Fable/issues/2703

Here are a few sample projects to get you started:
- https://github.com/ncave/fable-raytracer
- https://github.com/ncave/fable-nbody
- https://github.com/ncave/fable-rust-sample

Feel free to ask questions, I can create simpler starter projects if needed (with or without std, if you are interested).
Have a great day!
